lib/init/index.js

```diff
@@ -1,65 +1,152 @@
+ * @param {String} action - Action to be done on the given ast
+ * @returns {Promise} - A promise that writes each transform, runs prettier
+ * and writes the file
  */
 
-function creator(options) {
-	let env = yeoman.createEnv("webpack", null);
-	const generatorName = options
-		? replaceGeneratorName(path.basename(options[0]))
-		: "webpack-default-generator";
-	if (options) {
-		const WebpackGenerator = class extends Generator {
-			initializing() {
-				options.forEach(path => {
-					return this.composeWith(require.resolve(path));
-				});
-			}
-		};
-		env.registerStub(WebpackGenerator, generatorName);
-	} else {
-		env.registerStub(defaultGenerator, "webpack-default-generator");
-	}
+const transformsObject = {
+	entryTransform,
+	outputTransform,
+	contextTransform,
+	resolveTransform,
+	devtoolTransform,
+	targetTransform,
+	watchTransform,
+	watchOptionsTransform,
+	externalsTransform,
+	nodeTransform,
+	performanceTransform,
+	statsTransform,
+	amdTransform,
+	bailTransform,
+	cacheTransform,
+	profileTransform,
+	moduleTransform,
+	pluginsTransform,
+	topScopeTransform,
+	mergeTransform,
+	devServerTransform,
+	modeTransform,
+	parallelismTransform,
+	recordsInputPathTransform,
+	recordsOutputPathTransform,
+	recordsPathTransform,
+	resolveLoaderTransform
+};
 
-	env.run(generatorName).on("end", _ => {
-		if (generatorName !== "webpack-default-generator") {
-			//HACK / FIXME
-			env = env.options.env;
-			return runTransform(env.configuration);
-		} else {
-			return runTransform(env.getArgument("configuration"));
-		}
+module.exports = function runTransform(webpackProperties, action) {
+	// webpackOptions.name sent to nameTransform if match
+	const webpackConfig = Object.keys(webpackProperties).filter(p => {
+		return p !== "configFile" && p !== "configPath";
 	});
-}
+	webpackConfig.forEach(scaffoldPiece => {
+		const config = webpackProperties[scaffoldPiece];
+		const transformations = Object.keys(transformsObject)
+			.map(k => {
+				const stringVal = k.substr(0, k.indexOf("Transform"));
+				if (Object.keys(config.webpackOptions).length) {
+					if (config.webpackOptions[stringVal]) {
+						return [transformsObject[k], config.webpackOptions[stringVal]];
+					} else {
+						return [transformsObject[k], config[stringVal]];
+					}
+				} else {
+					return [transformsObject[k]];
+				}
+			})
+			.filter(e => e[1]);
+		const ast = j(
+			action && action !== "init"
+				? webpackProperties.configFile
+				: "module.exports = {}"
+		);
+		const transformAction = action || null;
 
-/*
-* @function replaceGeneratorName
-*
-* Replaces the webpack-addons pattern with the end of the addons name merged
-* with 'generator'
-*
-* @param { String } name - name of the generator
-* @returns { String } name - replaced pattern of the name
-*/
+		return pEachSeries(transformations, f => {
+			if (!f[1]) {
+				return f[0](j, ast, transformAction);
+			} else {
+				return f[0](j, ast, f[1], transformAction);
+			}
+		})
+			.then(_ => {
```

Comment:
`_ => {` can be `() => {`

Reply:
Noop functions are `_` Google/Chrome convention that is persistent across the source code

Reply:
But this is not a NOOP function

Reply:
yeah it is kinda misleading provided we will also have lodash defined in `_`

Reply:
It is rather inconsistent 
![image](https://user-images.githubusercontent.com/1002461/37202036-b314862a-2389-11e8-85b2-2975cb011a50.png)
I think @ev1stensberg is referring to the common practice in functional programming of using `_` when the parameter is not used

Reply:
Ok!